enhancements/proxy-cache.md

```diff
@@ -0,0 +1,124 @@
+  * The pulled layers are saved to cache and served to the user in parallel.
+  * This is depicted as below:
+    ![](https://user-images.githubusercontent.com/11522230/145871778-da01585a-7b1b-4c98-903f-809c214578da.png)
+    Design credits: @fmissi
+
+2. If the image exists in cache:
+  * A user can rely on Quay's cache to stay coherent with the upstream source so that I transparently get newer images from the cache
+     when tags have been overwritten in the upstream registry immediately or after a certain period of time.
+  * If the upstream image and cached version of the image are same:
+    * No layers are pulled from the upstream repository and the cached image is served to the user.
+
+  * If the upstream image and cached version of the image are different:
+    * The user is authenticated into the upstream registry and only the changed layers of `postgres:14` are pulled.
+    * The new layers are updated in cache and served to the user in parallel.
+  * This is depicted as below:
+    ![](https://user-images.githubusercontent.com/11522230/145872216-31350e08-6746-4e34-aebf-e59a7bf6b372.png)
+    Design credits: @fmissi
+
+3. If user initiates a pull when the upstream registry which is down:
+  * If the pull happens with the configured staleness period, the image stored in cache is served.
+  * If the pull happens after the configured staleness period, the error is propagated to the user.
+  * This is depicted as below:
+    ![](https://user-images.githubusercontent.com/11522230/145878373-c23d094b-709d-4859-b875-013ea33e34f7.png)
+    Design credits: @fmissi
+
+A quay admin can leverage quota of an organization to limit cache size so the backend storage consumption remains predictable
+by discarding images from the cache according to least recently used frequency.
+This is depicted as below:
+![](https://user-images.githubusercontent.com/11522230/145884935-df19297f-96b5-4c1c-9cdc-e199e04df176.png)
+Design credits: @sdadi
+
+A user initiates a pulls of an image(say a `postgres:14` image) from an upstream repository on Quay. If the storage consumption of the organization
+is beyond the quota limit, images in the namespace are removed based on least recently used, to make space for `postgres:14` to be cached.
+
+### Constraints
+
+* If quota management is enabled with proxy cache organization, and say an org is set to a max quota of 500mb and the image the user wants to pull is 700mb.
+  In such a case, the image pulled will be cached and will overflow beyond quota limit.
+
+### Risks and Mitigations
+
+* The cached images should have all properties that images on a Quay repository would have.
+* When an upstream image exists in cache, only the updated layers are to be pulled.
+
+### Implementation Plan
+
+* Pull a fresh image and check that layers are saved to cache.
+* Pull an outdated image and check that only changed layers are saved to cache.
+* Implement configuring of a proxy cache organization.
+* Implement quota management.
```

Comment:
Same comment as above as to what "quota" means in this context.

Reply:
We do depend on quota reporting, since that will introduce the repository size calculations. The pull thry proxy [epic](https://issues.redhat.com/browse/PROJQUAY-465) is currently "blocked by" the quota epic. It doesn't hurt to mention it here though.

Reply:
I've added the dependency here.